File: src/main/java/com/owncloud/android/ui/activities/data/files/FilesRepository.java
/**
 *   Nextcloud Android client application
 *
 *   Copyright (C) 2018 Edvard Holst
 *
 *   This program is free software; you can redistribute it and/or
 *   modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
 *   License as published by the Free Software Foundation; either
 *   version 3 of the License, or any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU AFFERO GENERAL PUBLIC LICENSE for more details.
 *
 *   You should have received a copy of the GNU Affero General Public
 *   License along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.owncloud.android.ui.activities.data.files;

import com.owncloud.android.datamodel.OCFile;
import com.owncloud.android.ui.activity.BaseActivity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Main entry point for accessing remote files
 */
public interface FilesRepository {
    interface ReadRemoteFileCallback {
        void onFileLoaded(@Nullable OCFile ocFile);
        void onFileLoadError(String error);
    }

    void readRemoteFile(String path, BaseActivity activity, @NonNull ReadRemoteFileCallback callback);
}

File: app/src/main/java/com/owncloud/android/ui/activities/data/files/FilesRepository.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2026 Alper Ozturk <alper.ozturk@nextcloud.com>
 * SPDX-FileCopyrightText: 2018 Edvard Holst <edvard.holst@gmail.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.ui.activities.data.files

import com.owncloud.android.datamodel.OCFile
import com.owncloud.android.ui.activity.BaseActivity

interface FilesRepository {
    interface ReadRemoteFileCallback {
        fun onFileLoaded(ocFile: OCFile?)
        fun onFileLoadError(error: String)
    }

    fun readRemoteFile(path: String, activity: BaseActivity, callback: ReadRemoteFileCallback)
}
